src/engine/source/cmds/src/test.cpp

```diff
@@ -1,423 +1,460 @@
+    if (parameters.jsonFormat)
     {
-        std::shared_ptr<store::FileDriver> driver;
-        json::Json testPolicy;
+        std::cout << jsonOutputAndTrace << std::endl;
+    }
+    else
+    {
+        printJsonAsYML(jsonOutputAndTrace);
+    }
+}
 
-        std::variant<json::Json, base::Error> get(const base::Name& name) const
+void run(std::shared_ptr<apiclnt::Client> client, const Parameters& parameters)
+{
+    using RequestTypeList = eTest::SessionsGet_Request;
+    using ResponseTypeList = eTest::SessionsGet_Response;
+    const std::string commandList {api::test::handlers::TEST_GET_SESSIONS_LIST_API_CMD};
+
+    // Check that the session exists before executing the run command
+    RequestTypeList eRequestList;
+    const auto requestList = utils::apiAdapter::toWazuhRequest<RequestTypeList>(commandList, details::ORIGIN_NAME, eRequestList);
+    const auto responseList = client->send(requestList);
+    const auto eResponseList = utils::apiAdapter::fromWazuhResponse<ResponseTypeList>(responseList);
+
+    auto foundSession {false};
+    for (const auto& session : eResponseList.list())
+    {
+        if (parameters.sessionName == session)
         {
-            if ("policy" == name.parts()[0])
-            {
-                return testPolicy;
-            }
-            else
-            {
-                return driver->get(name);
-            }
+            foundSession = true;
         }
-    };
-    auto _testDriver = std::make_shared<TestDriver>();
-    _testDriver->driver = fileStore;
-    _testDriver->testPolicy = json::Json {envTmp};
-
-    // TODO: Handle errors on construction
-    builder::Builder _builder(_testDriver, registry);
-    decltype(_builder.buildPolicy({options.policy})) env;
-    try
-    {
-        env = _builder.buildPolicy({options.policy});
     }
-    catch (const std::exception& e)
+
+    if (!foundSession)
     {
-        LOG_ERROR("Engine 'test' command: An error occurred while building the policy '{}': {}.",
-                  options.policy,
-                  utils::getExceptionStack(e));
-        g_exitHanlder.execute();
+        std::cout <<  fmt::format("Session '{}' could not be found", parameters.sessionName) << std::endl;
         return;
     }
 
-    // Create rxbackend
-    auto controller = rxbk::buildRxPipeline(env);
-    g_exitHanlder.add([&controller]() { controller.complete(); });
+    // Call run command
+    using RequestType = eTest::RunPost_Request;
+    using ResponseType = eTest::RunPost_Response;
+    const std::string command {api::test::handlers::TEST_RUN_API_CMD};
 
-    // output
-    std::stringstream output;
-    auto stderrSubscriber = rxcpp::make_subscriber<rxbk::RxEvent>(
-        [&](const rxbk::RxEvent& event) { output << event->payload()->prettyStr() << std::endl; });
-    controller.getOutput().subscribe(stderrSubscriber);
+    using RequestType = eTest::RunPost_Request;
+    using ResponseType = eTest::RunPost_Response;
```

Comment:
duplicate lines

Reply:
I am sorry, I don't see the duplicated lines